types/command-line-args/index.d.ts

```diff
@@ -1,14 +1,18 @@
 // Type definitions for command-line-args 5.0
 // Project: https://github.com/75lb/command-line-args
-// Definitions by: Lloyd Brookes <https://github.com/75lb>
+// Definitions by: Lloyd Brookes <https://github.com/75lb>,
+//                 Emanuele Stoppa <https://github.com/ematipico>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.2
 
 /**
  * Returns an object containing option values parsed from the command line. By default it parses the global `process.argv` array.
  * Parsing is strict by default. To be more permissive, enable `partial` or `stopAtFirstUnknown` modes.
  */
-declare function commandLineArgs(optionDefinitions: commandLineArgs.OptionDefinition[], options?: commandLineArgs.ParseOptions): commandLineArgs.CommandLineOptions;
+declare function commandLineArgs(
+    optionDefinitions: commandLineArgs.OptionDefinition[] | Array<object & commandLineArgs.OptionDefinition>,
```

Comment:
Hmm.. I'm a bit confused here. The original definition should accept a `CustomDefinition[]` too. If this isn't the case can you share the error you're seeing (and the code)?

Reply:
I don't have the code right now as it's on a different machine of mine.
But I clearly remember that when trying doing:

```js
const cmdArgs = require('command-line-args');
const command = {
        name: 'something',
        type: String,
        defaultValue: '1',
        lazyMultiple: true,
        defaultOption: true,
        group: 'one',
        description: 'Desc' // where description is a new property which is correctly accepted by the library
}
cmdArgs([command]) // <== here error
```

Was throwing an error saying that `command` should be of type `commandLineArgs.OptionDefinition`. 

My first attempt of fixing the interface was to use a generic that extends `commandLineArgs.OptionDefinition` but the linter was throwing an error